www/js/diary/timelineHelper.ts

```diff
@@ -217,3 +222,349 @@ const locations2GeojsonTrajectory = (trip, locationList, trajectoryColor?) => {
     };
   });
 };
+
+// Remaining functions from /diary/services.js
+const unpackServerData = (obj: ServerData<any>) => ({
+  ...obj.data,
+  _id: obj._id,
+  key: obj.metadata.key,
+  origin_key: obj.metadata.origin_key || obj.metadata.key,
+});
+
+export const readAllCompositeTrips = function (startTs: number, endTs: number) {
+  const readPromises = [getRawEntries(['analysis/composite_trip'], startTs, endTs, 'data.end_ts')];
+  return Promise.all(readPromises)
+    .then(([ctList]: [ServerResponse<CompositeTrip>]) => {
+      return ctList.phone_data.map((ct) => {
+        const unpackedCt = unpackServerData(ct);
+        return {
+          ...unpackedCt,
+          start_confirmed_place: unpackServerData(unpackedCt.start_confirmed_place),
+          end_confirmed_place: unpackServerData(unpackedCt.end_confirmed_place),
+          locations: unpackedCt.locations?.map(unpackServerData),
+          sections: unpackedCt.sections?.map(unpackServerData),
+        };
+      });
+    })
+    .catch((err) => {
+      displayError(err, 'while reading confirmed trips');
+      return [];
+    });
+};
+
+const dateTime2localdate = function (currtime: DateTime, tz: string) {
+  return {
+    timezone: tz,
+    year: currtime.get('year'),
+    //the months of the draft trips match the one format needed for
+    //moment function however now that is modified we need to also
+    //modify the months value here
+    month: currtime.get('month') + 1,
+    day: currtime.get('day'),
+    weekday: currtime.get('weekday'),
+    hour: currtime.get('hour'),
+    minute: currtime.get('minute'),
+    second: currtime.get('second'),
+  };
+};
+
+const points2TripProps = function (locationPoints) {
+  const startPoint = locationPoints[0];
+  const endPoint = locationPoints[locationPoints.length - 1];
+  const tripAndSectionId = `unprocessed_${startPoint.data.ts}_${endPoint.data.ts}`;
+  const startTime = DateTime.fromSeconds(startPoint.data.ts).setZone(startPoint.metadata.time_zone);
+  const endTime = DateTime.fromSeconds(endPoint.data.ts).setZone(endPoint.metadata.time_zone);
+
+  const speeds = [],
+    dists = [];
+  var loc, locLatLng;
+  locationPoints.forEach((pt) => {
+    const ptLatLng = L.latLng([pt.data.latitude, pt.data.longitude]);
+    if (loc) {
+      const dist = locLatLng.distanceTo(ptLatLng);
+      const timeDelta = pt.data.ts - loc.data.ts;
+      dists.push(dist);
+      speeds.push(dist / timeDelta);
+    }
+    loc = pt;
+    locLatLng = ptLatLng;
+  });
+
+  const locations = locationPoints.map((point, i) => ({
+    loc: {
+      coordinates: [point.data.longitude, point.data.latitude],
+    },
+    ts: point.data.ts,
+    speed: speeds[i],
+  }));
+
+  // used to mimic old momentJS moment.format()
+  const formatString = "yyyy-MM-dd'T'HH:mm:ssZZ";
```

Comment:
I'm pretty sure Luxon's `toISO()` is the same as moment's default `format()`.
If so, it would be a cleaner way of going about this than typing this out